whisper.cpp

```diff
@@ -4702,17 +4963,17 @@ static void whisper_process_logits(
         // ref: https://github.com/openai/whisper/blob/0b1ba3d46ebf7fe6f953acfd8cad62a4f851b49f/whisper/decoding.py#L431-L437
         {
             // logsumexp over timestamps
+            // ref: https://pytorch.org/docs/stable/generated/torch.logsumexp.html
             float timestamp_logprob = -INFINITY;
             {
                 float logsumexp = 0.0f;
-                const float logprob_max = *std::max_element(logprobs.begin() + vocab.token_beg, logprobs.end());
                 for (int i = vocab.token_beg; i < n_logits; ++i) {
                     if (logprobs[i] > -INFINITY) {
-                        logsumexp += expf(logprobs[i] - logprob_max);
+                        logsumexp += expf(logprobs[i]);
                     }
                 }
                 if (logsumexp > 0.0f) {
-                    timestamp_logprob = logf(logsumexp) + logprob_max;
+                    timestamp_logprob = logf(logsumexp);
                 }
```

Comment:
This change is not necessary - the old implementation is a more numerically stable way of computing the `logsumexp`

Reply:
OK, I will revert that.